slider/curve.py

```diff
@@ -0,0 +1,59 @@
+import math
+from .position import Position
+
+class Bezier():
+    def __init__(self, points):
+        self.points = points
+        self.xs = [i.x for i in points]
+        self.ys = [i.y for i in points]
+
+    def at(self, t):
+        def f(points, t):
+            if len(points) == 1:
+                return points[0]
+            return (1 - t) * f(points[:-1], t) + t * f(points[1:], t)
+        return Position(f(self.xs, t), f(self.ys, t))
+
+class Passthrough():
+    def __init__(self, points):
+        self.points = points
+        self.centre = getCentre(*self.points)
+        
+        a = [math.atan2(p.y - self.centre.y, p.x - self.centre.x) for p in points]
+        # angles of 3 points
+        
+        self.angle = a[2] - a[0]
+        """angle of arc sector that describes slider"""
+
+        # switch angle direction if necessary
+        if a[1] - a[0] < a[2] - a[0]:
+            self.angle *= -1
+
+    def at(self, t):
+        return rotate(self.points[0], self.centre, self.angle * t)
+
+def getCentre(p1, p2, p3):
+    """Returns the Position of the centre of the circle described by the 3 points"""
+    if p2.x == p1.x:
+        # normal t calc won't work
+        t = (p3.y - p1.y) / (2 * (p3.x - p2.x))
+        cx = 0.5 * (p2.x + p3.x) + t * (p3.y - p2.y)
+        cy = 0.5 * (p2.y + p3.y) - t * (p3.x - p2.x)
+    else:
+        if p3.y == p2.y:
+            t = (p3.x - p1.x) / (2 * (p2.y - p1.y))
+        else:
+            t = ((-(p1.y - p3.y) / (2 * (p2.x - p1.x)))
```

Comment:
Unfortunately this really is just a horrible expression.  
It's the value of t you get by solving the parametric equations for the perpendicular bisectors of the chords made by the points as simultaneous equations. Basically finding where they cross.  
If you can find a nicer way of writing it, I'm all for it but I think it's guaranteed to be ugly because of all the coordinate differences you have to do.